packages/sources/blocksize-capital/src/transport/vwap.ts

```diff
@@ -0,0 +1,70 @@
+import { BaseEndpointTypes } from '../endpoint/vwap'
+import { WebsocketReverseMappingTransport } from '@chainlink/external-adapter-framework/transports/websocket'
+import { makeLogger } from '@chainlink/external-adapter-framework/util'
+import {
+  BaseMessage,
+  VwapUpdate,
+  blocksizeDefaultUnsubscribeMessageBuilder,
+  blocksizeDefaultWebsocketOpenHandler,
+  buildBlocksizeWebsocketTickersMessage,
+  handlePriceUpdates,
+} from './utils'
+
+const logger = makeLogger('BlocksizeCapitalVwapWebsocketEndpoint')
+
+// On subscription to Blocksize-Capital WS API, the initial response is sent as a
+// snapshot (in result.snapshot) of the most recent data.
+// Subsequent responses are sent as notifications in params.updates
+export interface FixedVwapMessage extends BaseMessage {
+  method?: 'fixedvwap'
+  params?: {
+    updates: VwapUpdate[]
+  }
+  result?: {
+    snapshot: VwapUpdate[]
+  }
+}
```

Comment:
Could we instead have two types, which should make `preProcessFixedVwapMessage` clearer with regards to types and avoid needing everything to be optional?

Something like

```suggestion
type Snapshot = {
    params: {
        updates: VwapUpdate[]
    }
}

type Update = {
    result: {
        snapshot: VwapUpdate[]
    }
}

type FixedVwapMessage = Snapshot | Update


```

Reply:
this looks nice, yeah will test it out